smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java

```diff
@@ -74,12 +74,26 @@ public static SMSHandler getInstance() {
     /**
      * Setup for the handler.
      *
-     * @param context current context.
-     */
-    public void setup(Context context) {
+     * @param context current context. Must not be null
+     * @throws NullPointerException If the given context is null
+     */
+    public void setup(@NonNull Context context) {
+        /*
+        null pointer exception is thrown because if the
+        context is null, this.context isn't, it's a WeakReference to a null object,
+        which later on causes problems
+         */
+        if(context == null) throw new NullPointerException("The given context must not be null!");
         this.context = new WeakReference<>(context);
     }
 
+    /**
+     * Clears the current context. Function used to test functionality
+     */
+    protected void clearSetup(){
```

Comment:
The thing is I don't need to set a different context, I need to clear the oneI have, and I can't do it using `setup()` because of the check that I added and because it's created a weakreference to a null object, while I need the context to be null

Reply:
The only methods in `SMSHandlerTest` which need to have `instance.context` point to a `null` value before they're run are `sendMessageWithNullContext_throws()` and `setupNullContext_throws()`. What you could do is delete method `clearSetup()`, and keep test methods in their current order, this way the only assignation of a non-null value to `instance.context` will be performed by `validMessage_isSent()` and `nullMessage_throws()`.

Reply:
Yup, as you saw bad practice having them in order.
For now I'll leave this as is. But there's something which might help this.
There's an issue which I think you opened regarding WeakReference. If that WeakReference gets removed, the function I use only for tests will become useless, so it will get dropped as a consequence.